cli/cmd/status.go

```diff
@@ -0,0 +1,107 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/runconduit/conduit/cli/healthcheck"
+
+	"github.com/runconduit/conduit/cli/k8s"
+	"github.com/runconduit/conduit/cli/shell"
+	pb "github.com/runconduit/conduit/controller/gen/public"
+	"github.com/spf13/cobra"
+)
+
+const lineWidth = 52
+
+var statusCmd = &cobra.Command{
+	Use:   "status",
+	Short: "Check your Conduit installation for potential problems.",
+	Long: `Check your Conduit installation for potential problems. The status command will perform various checks of your
+local system, the Conduit control plane, and connectivity between those. The process will exit with non-zero status if
+problems were found.`,
+	Args: cobra.NoArgs,
+	Run: exitSilentlyOnError(func(cmd *cobra.Command, args []string) error {
+
+		kubeApi, err := k8s.MakeK8sAPi(shell.MakeUnixShell(), kubeconfigPath, apiAddr)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error: %s\n", err.Error())
+			return statusCheckResultWasError(os.Stdout, err)
+		}
+
+		client, err := newApiClient(kubeApi)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error: %s\n", err.Error())
+			return statusCheckResultWasError(os.Stdout, err)
+		}
+
+		kubectl, err := k8s.MakeKubectl(shell.MakeUnixShell())
```

Comment:
perhaps for a subsequent PR, but assuming `kubectl` is just one in a list of `StatusChecker`'s, should we instead fail this particular checker and carry on with other checks?

Reply:
That's a good question. I like being able to do as much as possible, so I think it is definitely something we'd want for next PRsa, especially if we do add copy-and-pastable stuff users can post when they report a bug.